src/cps/search.cpp

```diff
@@ -366,13 +367,21 @@ namespace cps::search {
 
         result.version = root->data.package.version.value_or("unknown");
 
+        const auto prefix_path = [&]() -> std::optional<fs::path> {
+            if (prefix_variable) {
+                return fs::path{prefix_variable.value()};
+            }
+            return std::nullopt;
+        }();
```

Comment:
I miss the c++23 monadic operators, `and_then(fs::path)` is so nice and clean

Reply:
me too... I was tempted to write a template for this but then did this instead

Reply:
The same author that wrote the expected implementation I used has an optional implementation that adds the same kinds of helpers... that is tempting

Reply:
I've previously vendored tl::optional for these functions in projects I've worked on.